### ISSUE TITLE
[Issue] Incompatibility with Pollution of the Realms' Pollution Blocks and NTM's Missiles

### ISSUE THREAD

**Issue by gatoborrachon** [Drillgon200/Hbm-s-Nuclear-Tech-GIT#164]
**What is happening**
When you launch a NTM Rocket and it goes through some Pollution of the Realms' Pollution Blocks, it will explode instead of ignore them

**Steps to reproduce**
1. Place some Pollution Blocks (in the example below i used Sulfur) on a specific area
2. Setup a rocket launch base and make it go through these Pollution Blocks
3. See it explode

**Screenshots**
![2021-10-26_10 19 39](https://user-images.githubusercontent.com/61126174/138912097-d4cebf71-148c-4a8c-b167-d2d67ccc0c26.png)
![2021-10-26_10 19 43](https://user-images.githubusercontent.com/61126174/138912107-3b9822cf-9826-4c92-a66a-052f134d8048.png)
![2021-10-26_10 19 44](https://user-images.githubusercontent.com/61126174/138912116-80bd86d9-f209-44d8-aad0-7a0baa5d38c9.png)

**Versions**
NTM: [hbm-1.6.4a-Gv1](https://www.curseforge.com/minecraft/mc-mods/hbms-nuclear-tech-mod-reloaded/files/3492216)
Pollution of the Realms: [1.12.2-1.2.21.1-build.0617](https://www.curseforge.com/minecraft/mc-mods/pollution-of-the-realms/files/3464562)
ForgeEndertech: [1.12.2-4.5.5.0-build.0561](https://www.curseforge.com/minecraft/mc-mods/forgeendertech/files/3137504) (Library of Pollution of the Realms)

**Additional Notes**
Well, there was a similar issue with the cars from Immersive Vehicles where they interact with these Pollution Blocks, but the Author of Pollution of the Realms told him [how to fix it and it worked](https://www.curseforge.com/minecraft/mc-mods/pollution-of-the-realms?comment=940), so i just wanted to tell you about this, to not let issues forgotten

In short, the fix only requires to add this check:
`state.getCollisionBoundingBox(world, pos) != null`
On lines like this one:
https://github.com/Drillgon200/Hbm-s-Nuclear-Tech-GIT/blob/master/com/hbm/entity/missile/EntityMissileBase.java#L228

**Comment by Drillgon200** [comment]
Didn't this one already get reported? That's not how I would fix it anyway; the missiles should be doing a real ray trace, not a crappy block check that allows it to skip through blocks.

**Comment by gatoborrachon** [comment]
well, yeah it was reported in the Discord server, but as you said one time there, it is easier for the dev to fix issues that are well documented and listed in the Github's Issues Section (for formality and readability)

And yeah, you also said that the missiles should do a raytrace, bad from my part, i forgot it but i also wanted to list here what the author of the other mod said. (and a reason why i added that just as an additional note, low importance)
